custom/icds_reports/tests/agg_tests/__init__.py

```diff
@@ -27,8 +28,20 @@
 OUTPUT_PATH = os.path.join(os.path.dirname(__file__), 'outputs')
 
 
+def _skip_if_nose_attr():
+    # When nose `attr` option is passed in tests
+    #   the setUpModule and tearDownModule are called even if it's
+    #   not necessary. This fails and further tests don't get run
+    # Since there is no icds tests with es_test attr, we can skip this
+    #   setUpModule and tearDownModule
+    return '--attr=es_test' in sys.argv
```

Comment:
There are other ways of passing the `--attr` option, which means this check is not very reliable. For example, it could be `-a es_test`. I'm not sure if there is a way to query the nose test runner to find out which attrs have been set, but that would be better.

Also, it's possible to specify multiple attributes to select a complex set of tests to be run, which may include tests that require the setup and teardown done in this module. Therefore, it may be a bad idea to skip setup/teardown even when the `es_test` attribute is specified. I'm not even sure if `_skip_if_nose_attr` is necessary the way it's currently used since nose should not run setup and teardown unless tests in this module are selected to be run.

Reply:
> I'm not even sure if _skip_if_nose_attr is necessary the way it's currently used since nose should not run setup and teardown unless tests in this module are selected to be run.

This has been what's happening without `_skip_if_nose_attr`, even though they weren't getting selected the `setUpModule` and `tearDownModule` were running.

I agree with the other concerns. Right now we are using the attr only in this ES test matrix where I am passing it as `--atr`, it's not being used in our code-base at all. It could be a concern if someone actually tried to use in a test matrix, which is less likely and hence this is okay?

Otherwise, I ended up spending a lot of time on this nose behaviour which I couldn't find a solution to. It's easily reproducible, you could simply decorate any one test with attr and run all tests by passing that decorator (even locally)

Reply:
> even though they weren't getting selected the setUpModule and tearDownModule were running

Hmm, that seems unexpected to me. I wonder if there's something about how these setup and teardown functions are written or used (not related to this PR) that is causing them to be called unnecessarily? No need to fix that there, and this seems good enough for now.